app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -165,8 +165,14 @@ def encryption_opts
     query_params = UriService.params(request.original_url)
     if query_params[:skip_encryption].present? && current_service_provider.skip_encryption_allowed
       nil
-    else
-      current_service_provider.encryption_opts
+    elsif current_service_provider.encrypt_responses?
+      cert = saml_request.service_provider.matching_cert || current_service_provider.ssl_certs.first
```

Comment:
So in updating the gem to try multiple certs... I figured maybe we just have the gem keep track of which one for us... it's a nasty side effect but I am getting desperate:
https://github.com/18F/saml_idp/pull/39/files#diff-29ac8f422d334e3198b369e7cd6d2139af467645af1128deb0def18de1c05409R27

Reply:
I think it's reasonable technical debt to take on - the gem is already a mess and tbh since the signature validation all happens there it feels like it will be cleaner.

Reply:
agreed